src/patient/PatientFiles.js

```diff
@@ -7,119 +7,127 @@ import TableHead from '@mui/material/TableHead';
-            aria-labelledby="modal-modal-title"
-            aria-describedby="modal-modal-description"
-          >
-            <Box sx={style}>
-              <Typography align="center" id="modal-modal-title" variant="h2" component="h2">
-                {`${data.file_name} | ${data.record_type}`}
-              </Typography>
-              <Box sx={{
-                display: 'grid',
-                gap: 1,
-                gridTemplateColumns: 'repeat(1fr)',
-              }}
-                component="form"
-              >
-                <Typography>{data.contents}</Typography>
-              </Box>
-            </Box>
-          </Modal>
-        </TableCell>
-      </TableRow>
-    ))
-    || (error && <TableRow><TableCell><Error message={error} /></TableCell></TableRow>)
-    || <TableRow><TableCell><CustomSpinner /></TableCell></TableRow>
-  }
+        (data && (data.record_type === sortTag || sortTag === '') && search(query, data) && (
+          <>
+            <TableRow
+              hover
+              onClick={handleOpen}
+              sx={{ '&:last-child td, &:last-child th': { border: 0 }, textDecoration: 'none' }}
+            >
+              <TableCell component="th" scope="row">
+                {data.file_name}
+              </TableCell>
+              <TableCell component="th" scope="row">
+                {data.record_type}
+              </TableCell>
+              <TableCell>{data.uploader_name}</TableCell>
+              <TableCell />
+              <TableCell>{data.date_uploaded ?? ''}</TableCell>
+            </TableRow>
+            <FileModal handleOpen={handleOpen} handleClose={handleClose} open={open} data={data} />
+          </>
+        ))
+        || (error && <TableRow><TableCell><Error message={error} /></TableCell></TableRow>)
+        || (
+          <TableRow style={{ height: '10vh' }}>
+            <TableCell align="center" />
+            <TableCell align="center" />
+            <TableCell align="left">{`No ${sortTag} Files`}</TableCell>
+            <TableCell align="center" />
+          </TableRow>
+
+        )
+      }
     </>
   );
 };
 
 const PatientFiles = (props) => {
   const { data } = useGetPatientInfoQuery();
+  const [sortTag, setSortTag] = useState('');
+  const [searchTerm, setSearchTerm] = useState('');
 
-  const tableContent = () => {
+  const tableContent = (filter, query) => {
     if (data) {
       // using index for id for now
-      return data.files.map((fileHash) => <PatientFile fileHash={fileHash} key={fileHash} />);
+      return data.files.map((fileHash) => <PatientFile query={query} sortTag={filter} fileHash={fileHash} key={fileHash} />);
     } else {
       // Temporary, will be replaced with an error component or not
       return <TableRow><TableCell><CustomSpinner /></TableCell></TableRow>;
     }
   };
 
+  const onQueryChange = (query) => {
+    console.log(query);
+    setSearchTerm(query);
+  };
+
+  const handleSort = (filter) => {
```

Comment:
Hehe we can just call setSortTag directly rather than putting it in this function. No need to change.